ckcp/openshift/base/deployment.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ckcp
+  labels:
+    app: kcp-in-a-pod
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kcp-in-a-pod
+  strategy:
+    type: Recreate
```

Comment:
Disruption may not matter for a development environment but the reason would be that we have embedded etcd. You can only have a single instance/process having the lock on the db files. With separate etcd processes, each etcd instance would have its own db files. They would form a quorum, elect a leader and ensure sync completion before acking a request. HA would be achieved in this way. This is not possible with a single embedded instance.

Reply:
@fgiloux so you're saying that since we have an embedded etcd (and not multiple like a HA environment), it's not possible for a new pod to sync to the same etcd? (in case of rolling deployment). Because at any point in time, its just one replica that exists.